src/vm/functions/mod.rs

```diff
@@ -347,3 +349,12 @@ fn special_as_contract(args: &[SymbolicExpression], env: &mut Environment, conte
 
     eval(&args[0], &mut nested_env, context)
 }
+
+fn special_principal_of(args: &[SymbolicExpression], env: &mut Environment, context: &LocalContext) -> Result<Value> {
```

Comment:
Do we need `principal-of` as a function? The trait reference resolves to a principal in any case (e.g., `(principal-of F)` is the same as just `F`

Reply:
Addressed with https://github.com/blockstack/blockstack-core/pull/1246/commits/c9f524a18b21fbd354bdd34a07f54a7ecd838d72